element-hq/element-x-android
# UX cleanup: DM details screen

## Type of change

- [x] Other : UX cleanup.

## Content

- Move send DM to a CTA button.
- Add 'Call' CTA button too when there is a DM with that user and a call is possible.
- Add missing tests.

## Motivation and context

Closes #2818.

## Screenshots / GIFs

|Before|After|
|-|-|
|![](https://media.githubusercontent.com/media/element-hq/element-x-android/0bbb107dea5f366967f3b7d1dc0e989bae81d8a0/tests/uitests/src/test/snapshots/images/ui_S_t%5Bf.userprofile.shared_UserProfileView_null_UserProfileView-Day-0_1_null_0%2CNEXUS_5%2C1.0%2Cen%5D.png)|![](https://media.githubusercontent.com/media/element-hq/element-x-android/bebc8556e82acc3a4d94d17a1807e6b6f8fb3d1a/tests/uitests/src/test/snapshots/images/ui_S_t%5Bf.userprofile.shared_UserProfileView_null_UserProfileView-Day-0_1_null_7%2CNEXUS_5%2C1.0%2Cen%5D.png)|

## Tests



- Go to a room with people you have some DMs with and others you don't.
- Select one of the people you DMed before, you should see 3 CTAs: Message, Call, Share. Call may be missing if the DM rules don't allow calls. Message should open the existing DM.
- Select a user you've never DMed. Message and Share should be available. Message should create a new DM with that user.
- Select your own profile. Only Share should be available.
- Now open a permalink to a user's profile. The same actions should be available depending on the relationship with that user.

## Tested devices

- OS version(s): 14

## Checklist
